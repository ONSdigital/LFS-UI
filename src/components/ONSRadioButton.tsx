import React, {CSSProperties} from 'react';

interface Props {
    label: string,
    id: string,
    onChange: ((...props: any[]) => void) | undefined,
    checked?: boolean,
    style?: CSSProperties
}

export const ONSRadioButton = (props: Props) => (
    <p className="radios__item" style={props.style}>
        <span className="radio">
            <input
                type="radio"
                id={props.id}
                className={"radio__input js-radio"}
                checked={props.checked}
                onChange={(e) => props.onChange ? props.onChange(e) : e}
                value={props.id}
                name={props.label}
            />
            <label className="radio__label " htmlFor={props.id}>
                {props.label}
            </label>
        </span>
    </p>
);
