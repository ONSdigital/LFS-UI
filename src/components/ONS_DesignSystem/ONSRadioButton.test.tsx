import React from 'react';
import Enzyme, {mount, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {ONSRadioButton} from "./ONSRadioButton";

describe("ONS Radio Button Test", () => {
    Enzyme.configure({adapter: new Adapter()});

    const radioProps = {
        label: "On/Off",
        id: '12',
    }

    const clickedRadioProps = {
        label: "On/Off",
        id: '12',
        onRadioClick: jest.fn()
    };

    function wrapper(render: any, props: any) {
        return render(
            <ONSRadioButton label={props.label} 
                            id={props.id} 
                            onChange={props.onRadioClick}/>)
    }

    it("should render correctly", () => expect(wrapper(shallow, radioProps).exists()).toEqual(true));

    it("should render with the correct label", () => {
        expect(wrapper(mount, radioProps).find("ONSRadioButton").getElement().props.label).toEqual(radioProps.label);
    });

    it('simulates change events', () => {
        wrapper(mount, clickedRadioProps).find('input').simulate('change');
        expect(clickedRadioProps.onRadioClick).toHaveBeenCalled()
    });
});