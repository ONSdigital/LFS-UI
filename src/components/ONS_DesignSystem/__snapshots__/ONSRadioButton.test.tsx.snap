// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ONS Radio Button Test matches Snapshot 1`] = `ShallowWrapper {}`;
